taxonomy/api/v1/views.py

```diff
@@ -0,0 +1,68 @@
+"""
+Taxonomy API views.
+"""
+from rest_framework import permissions
+from rest_framework.mixins import ListModelMixin, RetrieveModelMixin
+from rest_framework.viewsets import GenericViewSet
+
+from django.db.models import Prefetch
+
+from taxonomy.api.v1.serializers import JobPostingsSerializer, JobsListSerializer, SkillListSerializer
+from taxonomy.models import CourseSkills, Job, JobPostings, Skill
+
+
+class TaxonomyAPIViewSetMixin:
+    """
+    Taxonomy APIs ViewSet Mixing.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    throttle_scope = 'taxonomy-api-throttle-scope'
+
+
+class SkillViewSet(TaxonomyAPIViewSetMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
+    """
+    ViewSet to list and retrieve all Skills in the system.
+    """
+
+    serializer_class = SkillListSerializer
+
+    def get_queryset(self):
+        """
+        Get all the skills with prefetch_related objects
+        """
+        return Skill.objects.all().prefetch_related(
+            Prefetch(
+                'courseskills_set',
+                queryset=CourseSkills.objects.filter(is_blacklisted=False)
+            )
+        )
+
+
+class JobsViewSet(TaxonomyAPIViewSetMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
+    """
+    ViewSet to list and retrieve all Jobs in the system.
+    """
+    serializer_class = JobsListSerializer
+
+    def get_queryset(self):
+        """
+        Get all the jobs with prefetch_related objects
+        """
+        return Job.objects.all().prefetch_related(
+            'jobskills_set', 'jobskills_set__skill'
+        )
+
+
+class JobPostingsViewSet(TaxonomyAPIViewSetMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
```

Comment:
All the viewsets are inheriting from same parents. what if we move the parents in `TaxonomyAPIViewSetMixin` or any other pattern to avoid listing all the parent in each child class

Reply:
@muhammad-ammar Surely we can do that, But it will create coupling in all views for the future, For example, in future, if would like to add a new API to update skills we would have to decouple all of them again.